### PR TITLE
Swift/C++: Use instanceof in more places

### PR DESCRIPTION
**Pull request by erik-krogh**
Uses `instanceof` clauses in more classes as preparation for enabling a QL-for-QL query that suggests using `instanceof`. 

Some of this was written automatically, some was written manually.  

Evaluations:  
 - [CPP](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/PR-11630-0-cpp/reports): There is some performance noise, but I can't replicate locally (I tried to replicate the `cpp/badly-bounded-write` regression reported by stage-timings, there was nothing), so I think we're good?    
 - [Swift](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/PR-11630-7-swift__2/reports): Some more performance noise, but in a positive direction, so that's good? 🤷

### CODE REVIEW

File: cpp/ql/lib/experimental/semmle/code/cpp/semantic/SemanticExprSpecific.qll [hunk 1]
```diff
@@ -286,19 +286,7 @@ module SemanticExprConfig {
...
-  }
-
-  private class ValueNumberBound extends Bound {
-    IRBound::ValueNumberBound bound;
-
-    ValueNumberBound() { bound = this }
-
-    override string toString() { result = bound.toString() }
-  }
+  class Bound = IRBound::Bound;
```
Comment (MathiasVP) [inline cpp/ql/lib/experimental/semmle/code/cpp/semantic/SemanticExprSpecific.qll:289]: This change looks like it's more than just "using instanceof in more places", no?

Comment (MathiasVP) [reply cpp/ql/lib/experimental/semmle/code/cpp/semantic/SemanticExprSpecific.qll:289]: > It's in experimental, so I think the API change is OK.

That's true. However, we _hope_ that this library won't be confined to experimental for that much longer 🤞.

> (The only visible API change is that the exported Bound class has all the predicates from IRBound::Bound).

Yeah, it's the increased public API surfaces that I'm most afraid of. What do you say to this @rdmarsh2?

Comment (erik-krogh) [reply cpp/ql/lib/experimental/semmle/code/cpp/semantic/SemanticExprSpecific.qll]: I see.  
I went for the minimal change instead.

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**